stock_location_tank/models/stock_picking.py

```diff
@@ -0,0 +1,165 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    mrp_production_ids = fields.One2many(
+        comodel_name="mrp.production",
+        inverse_name="picking_id",
+        string="Manufacturing Orders",
+    )
+
+    def action_view_mrp_production(self):
+        self.ensure_one()
+        xmlid = "mrp.mrp_production_action"
+        action = self.env["ir.actions.act_window"]._for_xml_id(xmlid)
+        form = self.env.ref("mrp.mrp_production_form_view")
+        # action = self.env.ref('mrp.mrp_production_action').read()[0]
+        if len(self.mrp_production_ids) == 1:
+            action["views"] = [(form.id, "form")]
+            action["res_id"] = self.mrp_production_ids[0].id
+        else:
+            action["domain"] = [("id", "in", self.mrp_production_ids.ids)]
+            action["context"] = dict(self.env.context)
+            action["context"]["search_default_todo"] = False
+        return action
+
+    def button_validate(self):
+        for rec in self:
+            if rec.location_dest_id.is_tank:
+                rec.env.context = dict(rec.env.context)
+                rec.env.context["allow_duplicate"] = True
+        return super(StockPicking, self).button_validate()
+
+    def _action_done(self):
+        res = super(StockPicking, self)._action_done()
+        for rec in self:
+            # Juntar els move_line_ids que tinguin el mateix product_id i location_id
+            moves_lines = {}
+            for line in rec.move_line_ids_without_package:
+                key = (line.product_id.id, line.location_dest_id.id)
```

Comment:
You should be able to use the objects directly `key = (line.product_id, line.location_dest_id)` this way you'll save a browse call below